azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/ConnectionStringBuilder.java

```diff
@@ -8,179 +8,167 @@
+     * Set an endpoint which can be used to connect to the EventHub instance.
      *
-     * @param endpointAddress     namespace level endpoint. This needs to be in the format of scheme://fullyQualifiedServiceBusNamespaceEndpointName
-     * @param entityPath          Entity path. For eventHubs case specify - eventHub name.
-     * @param sharedAccessKeyName Shared Access Key name
-     * @param sharedAccessKey     Shared Access Key
+     * @param namespaceName the name of the namespace to connect to.
+     * @param domainName    identifies the domain the namespace is located in. For non-public and national clouds,
+     *                      the domain will not be "servicebus.windows.net". Available options include:
+     *                          - "servicebus.usgovcloudapi.net"
+     *                          - "servicebus.cloudapi.de"
+     *                          - "servicebus.chinacloudapi.cn"
+     * @return the {@link ConnectionStringBuilder} being set.
      */
-    public ConnectionStringBuilder(
-            final URI endpointAddress,
-            final String entityPath,
-            final String sharedAccessKeyName,
-            final String sharedAccessKey) {
-        this(endpointAddress, entityPath, sharedAccessKeyName, sharedAccessKey, MessagingFactory.DefaultOperationTimeout, RetryPolicy.getDefault());
+    public ConnectionStringBuilder setEndpoint(String namespaceName, String domainName) {
+        try {
+            this.endpoint = new URI(String.format(Locale.US, endpointFormat, namespaceName, domainName));
+        } catch (URISyntaxException exception) {
+            throw new IllegalConnectionStringFormatException(
+                    String.format(Locale.US, "Invalid namespace name: %s", namespaceName),
+                    exception);
+        }
+        return this;
     }
 
     /**
-     * Build a connection string consumable by {@link com.microsoft.azure.eventhubs.EventHubClient#createFromConnectionString(String, Executor)}
+     * Set a namespace name which will be used to connect to an EventHubs instance. This method adds
+     * "servicebus.windows.net" as the default domain name.
      *
-     * @param endpointAddress       namespace level endpoint. This needs to be in the format of scheme://fullyQualifiedServiceBusNamespaceEndpointName
-     * @param entityPath            Entity path. For eventHubs case specify - eventHub name.
-     * @param sharedAccessSignature Shared Access Signature
+     * @param namespaceName the name of the namespace to connect to.
+     * @return the {@link ConnectionStringBuilder} being set.
      */
-    public ConnectionStringBuilder(
-            final URI endpointAddress,
-            final String entityPath,
-            final String sharedAccessSignature) {
-        this(endpointAddress, entityPath, sharedAccessSignature, MessagingFactory.DefaultOperationTimeout, RetryPolicy.getDefault());
+    public ConnectionStringBuilder setNamespaceName(String namespaceName) {
```

Comment:
The old version of the ConnectionStringBuilder took a namespace as an argument, so that's how it's been in the java client at least. This method will generate a URI for the user just like the old java client did. We should talk about it more before merging.